src/util/safe_print.cpp

```diff
@@ -0,0 +1,200 @@
+/*********************                                                        */
+/*! \file safe_print.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andres Noetzli
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2016 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief [[ Add one-line brief description here ]]
+ **
+ ** [[ Add lengthier description here ]]
+ ** \todo document this file
+ **/
+
+#include "safe_print.h"
+
+#include <unistd.h>
+
+/* Size of buffers used */
+#define BUFFER_SIZE 20
+
+namespace CVC4 {
+
+template <>
+void safe_print(int fd, const char* msg) {
+  ssize_t ret = write(fd, msg, safe_strlen(msg));
+  (void)ret;
+}
+
+template <>
+void safe_print(int fd, const std::string* msg) {
+  ssize_t ret = write(fd, msg->c_str(), msg->length());
+  (void)ret;
+}
+
+template <>
+void safe_print(int fd, int64_t i) {
+  char buf[BUFFER_SIZE];
+
+  if (i == 0) {
+    safe_print(fd, "0");
+    return;
+  } else if (i < 0) {
+    safe_print(fd, "-");
+  }
+
+  size_t idx = BUFFER_SIZE - 1;
+  while (i != 0 && idx != 0) {
+    buf[idx] = '0' + i % 10;
+    i /= 10;
+    idx--;
+  }
+  ssize_t ret = write(fd, buf + idx + 1, BUFFER_SIZE - idx - 1);
+  (void)ret;
```

Comment:
Checking would be nice. But we need to have a sane semantics if it fails. Maybe kill the process if the size does not match the expected value?

If we keep the current code maybe add a comment about ignoring the value. maybe switch to static_cast<void>(write(...)); ?